src/hosted-form/iframe-content/hosted-input.ts

```diff
@@ -0,0 +1,206 @@
+import { kebabCase } from 'lodash';
+
+import { IframeEventListener, IframeEventPoster } from '../../common/iframe';
+import { InvalidHostedFormConfigError } from '../errors';
+import { HostedFieldEventMap, HostedFieldEventType, HostedFieldSubmitRequestEvent } from '../hosted-field-events';
+import HostedFieldType from '../hosted-field-type';
+
+import HostedInputAggregator from './hosted-input-aggregator';
+import { HostedInputEvent, HostedInputEventType } from './hosted-input-events';
+import HostedInputStyles, { HostedInputStylesMap } from './hosted-input-styles';
+import HostedInputValidator from './hosted-input-validator';
+import HostedInputWindow from './hosted-input-window';
+
+export default class HostedInput {
+    protected _input: HTMLInputElement;
+    protected _previousValue?: string;
+    private _isSubmitted: boolean = false;
+    private _isTouched: boolean = false;
+
+    /**
+     * @internal
+     */
+    constructor(
+        protected _type: HostedFieldType,
+        protected _containerId: string,
+        protected _placeholder: string,
+        protected _accessibilityLabel: string,
+        protected _autocomplete: string,
+        protected _styles: HostedInputStylesMap,
+        protected _eventListener: IframeEventListener<HostedFieldEventMap>,
+        protected _eventPoster: IframeEventPoster<HostedInputEvent>,
+        protected _inputAggregator: HostedInputAggregator,
+        protected _inputValidator: HostedInputValidator
+    ) {
+        this._input = document.createElement('input');
+
+        this._input.addEventListener('input', this._handleInput);
+        this._input.addEventListener('blur', this._handleBlur);
+        this._input.addEventListener('focus', this._handleFocus);
+        this._eventListener.addListener(HostedFieldEventType.SubmitRequested, this._handleSubmit);
+
+        this._configureInput();
+    }
+
+    getType(): HostedFieldType {
+        return this._type;
+    }
+
+    getValue(): string {
+        return this._input.value;
+    }
+
+    setValue(value: string): void {
+        this._processChange(value);
+    }
+
+    isTouched(): boolean {
+        return this._isTouched;
+    }
+
+    attach(): void {
```

Comment:
should the `containerId` be an argument of attach() instead of the constructor?

Reply:
I guess it could. But I don't really need it as an argument for my use case. Right not it's in the constructor and can't be changed for the lifetime of the instance. Potentially we could move it to `attach` if there's a need for it in the future.

Reply:
yeah the main reason was so that the code is more self documented `attach(containerId: string): void`